frontend/src/pages/BookPage.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import Catalog from '../components/Catalog';
+import BookInfo from '../components/BookInfo';
+import { getBookData, getQuizzes } from '../utils/api';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Button, Row } from 'reactstrap';
+class BookPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      bookID: props.match.params.id,
+      quizIndex: -1,
+      quizID: -1,
+      currentQuestions: []
+    };
+    this.fetchBookData();
+    this.fetchQuizData();
+  }
+
+  fetchBookData = async () => {
+    const { message, success, result } = await getBookData(
+      this.props.match.params.id
+    );
+    if (success) {
+      this.setState({ bookData: result['results'][0] });
```

Comment:
do we need some check to make sure the book data exists ex: result['results'] has length >= 1?